src/base/memory_monitor_sr.hpp

```diff
@@ -176,11 +176,10 @@ namespace cubmem
       {
 	MMON_METAINFO *metainfo = (MMON_METAINFO *) get_metainfo_pos (ptr, allocated_size);
 
-	if (metainfo->magic_number == m_magic_number)
+	if (metainfo->magic_number == m_magic_number && metainfo->allocated_size == allocated_size
+	    && metainfo->stat_id < MMON_MAP_RESERVE_SIZE && metainfo->stat_id >= 0)
```

Comment:
Nit. 
I can't understand what this condition is for just by looking at the code. How about giving some meaning to the conditions ? 
```suggestion
        const bool meaningful_name = metainfo->magic_number == m_magic_number 
        && metainfo->allocated_size == allocated_size
        && metainfo->stat_id < MMON_MAP_RESERVE_SIZE
        && metainfo->stat_id >= 0
        
	if (meaningful_name)
```

Reply:
Same opinion with @joohok.

Basically, this two conditions, allocated_size >= MMON_METAINFO_SIZE and metainfo->magic_number == m_magic_number, guarantee that this memory is traced by mmon. The reason why tests with different conditions are needed must be explained in comments.